gwlfe/WriteOutputFiles.py

```diff
@@ -7,9 +7,717 @@
 
 
 def WriteOutput(z):
-    print('WriteOutput')
+    # DIMENSION VARIABLES FOR PREDICT CALCULATION AND SCENARIO FILE
+    AvOtherLuSed = 0
+    AvOtherLuNitr = 0
+    AvOtherLuPhos = 0
+    TotSewerSys = 0
+    TotNormSys = 0
+    TotShortSys = 0
+    TotSeptSys = 0
+    TotAvLuErosion = 0
+    AvTotalSed = 0
+    AvDisN = 0
+    AvTotalN = 0
+    AvDisP = 0
+    AvTotalP = 0
+    n2t = 0
+    n6t = 0
+    n13t = 0
+    n24t = 0
+
+    z.SumTotArea = z.SumTotArea + z.TotArea
+
+    # INSERT VALUES FOR BMP SCENARIO FILE FOR PREDICT APPLICATION
+    for l in range(0, z.NLU):
+        z.AvLuSedYield[l] = (z.AvLuSedYield[l] * z.RetentFactorSed) * (1 - z.AttenTSS)
+        z.AvLuDisNitr[l] = (z.AvLuDisNitr[l] * z.RetentFactorN) * (1 - z.AttenN)
+        z.AvLuTotNitr[l] = (z.AvLuTotNitr[l] * z.RetentFactorN) * (1 - z.AttenN)
+        z.AvLuDisPhos[l] = (z.AvLuDisPhos[l] * z.RetentFactorP) * (1 - z.AttenP)
+        z.AvLuTotPhos[l] = (z.AvLuTotPhos[l] * z.RetentFactorP) * (1 - z.AttenP)
+
+    # SET THE SCENARIO VALUES TO LANDUSE LOADS
+    for l in range(0, z.NRur):
+        if (z.Landuse[l].lower() == "hay/past"):
+            z.n2 = z.AvLuSedYield[l]
+            z.n6 = z.AvLuTotNitr[l]
+            z.n13 = z.AvLuTotPhos[l]
+            z.n6dn = z.AvLuDisNitr[l]
+            z.n13dp = z.AvLuDisPhos[l]
+            z.n24 = round(z.Area[l])
+        elif (z.Landuse[l].lower() == "cropland"):
+            z.n1 = z.AvLuSedYield[l]
+            z.n5 = z.AvLuTotNitr[l]
+            z.n12 = z.AvLuTotPhos[l]
+            z.n5dn = z.AvLuDisNitr[l]
+            z.n12dp = z.AvLuDisPhos[l]
+            z.n23 = round(z.Area[l])
+        elif (z.Landuse[l].lower() == "turfgrass"):
+            z.n2t = z.AvLuSedYield[l]
+            z.n6t = z.AvLuTotNitr[l]
+            z.n13t = z.AvLuTotPhos[l]
+            z.n24t = round(z.Area[l])
+        elif (z.Landuse[l].lower() == "unpaved_road"):
+            z.n2d = z.AvLuSedYield[l]
+            z.n6d = z.AvLuTotNitr[l]
+            z.n13d = z.AvLuTotPhos[l]
+            z.n6ddn = z.AvLuDisNitr[l]
+            z.n13ddp = z.AvLuDisPhos[l]
+        else:
+            AvOtherLuSed = AvOtherLuSed + z.AvLuSedYield[l]
+            AvOtherLuNitr = AvOtherLuNitr + z.AvLuTotNitr[l]
+            AvOtherLuPhos = AvOtherLuPhos + z.AvLuTotPhos[l]
+
+    LoadReductions.AdjustScnLoads(z)
+
+    z.n2c = 0
+    z.n6c = 0
+    z.n13c = 0
+    z.n24b = 0
+    z.n2b = 0
+    z.n6b = 0
+    z.n13b = 0
+    z.n23b = 0
+    z.n6cdn = 0
+    z.n13cdp = 0
+    z.n6bdn = 0
+    z.n13bdp = 0
+
+    for l in range(z.NRur + 1, z.NLU):
```

Comment:
I think this should actually be `range(z.NRur, z.NLU)`. Indices 0-9 should be rural, so the first urban land use is at index 10.